docs/src/contributing/git-workflow.md

```diff
@@ -51,36 +99,173 @@ If the branch exists in a triton/neptune official repo, (as opposed to a persona
- - Update `README.md` in case to make sure the installation instructions are up-to-date.
+## deviation from standard semver practice
+
+neptune-core is presently both a (blockchain) library and a binary. Semantic versioning signals changes to APIs and mainly applies to libraries.  But it does not capture breaking changes that can occur in a blockchain ecosystem or breaking changes with regards to stored state, ie databases.
+
+For blockchains it is of primary importance to maintain consensus and network
+compatibility with other nodes on the network.
+
+For these reasons, we define the version identifier as follows.
+
+## version identifier
+
+neptune-core uses sem-ver compatible identifier of the form:
+```
+<major>.<minor>.<patch>[-<candidate>]
+
+where `candidate` is of the form "rc1", "rc2", etc and starts at 1.
+```
+
+Examples:
+```
+0.4.0-rc1
+0.5.0-rc1
+0.5.0-rc2
+0.5.0-rc3
+1.2.2
+```
+
+Non-Examples: (should not occur)
+```
+0.4.1-rc1       (point release should not be a release-candidate)
+1.2.0-rc0       (release candidates start at 1, not 0)
+```
+
+note: for git tags, a "v" is prefixed, as per git convention.
+
+Our release process treats these as:
+
+**major**: bumped any time there are:
+
+1. significant consensus or p2p layer changes. eg hard or soft-fork or breaking protocol change.
+2. breaking library API changes.
+3. breaking database changes.
+
+resets minor and patch to 0.
+
+**minor**: bumped any time changes from master are included in a release candidate.
+
+resets patch to 0.
+
+**patch**: bumped any time changes from a release branch are included in a release.
+
+resets candidate to "-rc1".
+
+**candidate**: added/bumped when a new public testnet is created for a candidate release.
+
+*Note that the optional suffix is supported by cargo and related tools.*
+
+## typical release
+
+The normal/typical release flow would be:
+
+1. normal changes (not consensus) are introduced in master.
+2. a release candidate is tagged from master, eg `v0.3.0-rc1` and built.
+3. a testnet is created for `v0.3.0-rc1`
+4. if significant problems are found then a new branch `v0.3.0-rc2` is created at `v0.3.0-rc1`, or possibly from master.  commit(s) are added. When ready, the final commit in the branch is tagged with `v0.3.0-rc2`, another testnet created, and so on.
+5. fixes for the release-candidate should also be applied to master, if applicable.
+6. After two weeks, or when team decides, the latest release-candidate tag is also tagged with `v0.3.0`, signifying an actual release.
+7. The release is performed.
+
+*note that by the time the release actually occurs, master may have diverged significantly.*
+
+The release flow for a major version change is essentially the same except that the major version number is bumped and the minor version returns to 0.
+
+
+## Release Candidate Protocol  (-rc1, -rc2, etc)
```

Comment:
I think some of these steps should be replaced by “follow the release protocol” and a link to said release protocol.

Reply:
by that, do you mean a link to docs/src/contributing/releasing.md?

Perhaps we need to ensure that document matches up with this one before merging this PR.  I haven't reviewed it yet, other than a 5 second scan just now.

Reply:
addressed in 66b171787d18c6ad4aaf45854923545d15666d28.

Ok, I made a first effort at syncing up these documents.  So now they link to eachother for some things rather than duplicating.   I think there is still some overlap and room to re-arrange some more, but I'd rather leave that to the author(s) of releasing.md and/or person(s) actually performing the releases.   That may be you @jan-ferdinand, I'm not certain actually.

Anyway there are some changes in releasing.md now that I'd appreciate if you can have a look at.